lessons/python_basics/IntroPython.ipynb

```diff
@@ -0,0 +1,1303 @@
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "s = 'This is a string.'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "T\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(s[0])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "orange\n"
+     ]
+    }
+   ],
+   "source": [
+    "# use -1 to get the last element\n",
+    "print(fruits[-1])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "lychee\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(fruits[-2])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "This\n"
+     ]
+    }
+   ],
+   "source": [
+    "# to get a slice of the string use the : symbol\n",
+    "print(s[0:4])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "This\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(s[:4])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      " is\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(s[4:7])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      " a string.\n",
+      " a string.\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(s[7:])\n",
+    "print(s[7:len(s)])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## If Statements"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "s2 is not a string or integer\n"
+     ]
+    }
+   ],
+   "source": [
+    "s2 = [19034, 23]\n",
+    "\n",
+    "# You will always need to start with an 'if' line\n",
+    "# You do not need the elif or else statements\n",
+    "# You can have as many elif statements as needed\n",
+    "\n",
+    "if type(s2) == str:\n",
```

Comment:
Better to use `isinstance(s2, str)`, but I guess it makes little difference for the example at hand.